v3/src/models/ui-state.ts

```diff
@@ -43,6 +43,13 @@ export class UIState {
   @observable
   private _interruptionCount = 0
 
+  @observable private _draggingDatasetId = ""
+  @observable private _draggingAttributeId = ""
+  @observable private _draggingXOffset = 0
+  @observable private _draggingYOffset = 0
+  @observable private _draggingOverlayHeight = 100
+  @observable private _draggingOverlayWidth = 100
```

Comment:
>- Information is passed from the API handler to the `PluginAttributeDrag` (`dataSetId` and `attributeId`) and `AttributeDragOverlay` (overlay size and position) via new `uiState` properties.
>   - This feels inelegant to me, and I am open to alternate methods of achieving this. Using the synthetic events won't work.


I thought about a couple of other options that might take more effort and/or might not work as well. In the end, I'm drawn to the simpler suggestion of what about making a new class, perhaps `DataInteractiveState`, and then putting these there?